client/asset/btc/btc.go

```diff
@@ -4188,6 +4188,99 @@ func (btc *baseWallet) createWitnessSig(tx *wire.MsgTx, idx int, pkScript []byte
 	return sig, privKey.PubKey().SerializeCompressed(), nil
 }
 
+// EstimateSendTxFee returns a tx fee estimate for sending or withdrawing the
+// provided amount using the provided feeRate.
+func (btc *baseWallet) EstimateSendTxFee(sendAmount, feeRate uint64, subtract bool) (fee uint64, err error) {
+	if sendAmount == 0 {
+		return 0, fmt.Errorf("cannot check fee: send amount = 0")
+	}
+
+	feeRate = btc.feeRateWithFallback(feeRate)
+
+	minTxSize := uint64(dexbtc.MinimumTxOverhead + btc.outputSize())
+
+	// If subtract is true, select enough inputs for sendAmount. Fees will be taken
+	// from the sendAmount. If not, select enough inputs to cover minimum fees.
+	enough := func(inputsSize, sum uint64) bool {
+		if subtract {
+			return sum >= sendAmount
+		}
+		minFee := (minTxSize + inputsSize) * feeRate
+		return sum >= sendAmount+minFee
+	}
+
+	btc.fundingMtx.RLock()
+	utxos, _, _, err := btc.spendableUTXOs(0)
+	btc.fundingMtx.RUnlock()
+	if err != nil {
+		return 0, err
+	}
+	totalIn, vSize, _, _, _, _, err := fund(utxos, enough)
```

Comment:
Is `sendtoaddress` going to pick the same inputs as `fund`?

Reply:
Yes, but the inputs picked here are not locked.

Reply:
Ah, glad I looked at this, I need to be using `btc.outputSize` in the `sendtoaddress` fix.

Reply:
> Yes, but the inputs picked here are not locked.

How to you know? When we use `Send` on an RPC wallet, we use the `sendtoaddress` call. Is bitcoind using the same prevout-picking algorithm as us?

Reply:
I don't think this point is resolved.  We're getting a very precise estimate for transaction with a certain number of inputs that does not reflect what an external wallet will do.

However, this is just an estimate so I don't think this is a big deal.  The main thing to me is that we get to see the current fee rate and subsequent work would probably make the fee rate on the Send form configurable.

That said, we can have different estimation code depending on the backend.  For instance, if the wallet type were `bitcoinRPC`, we could use `fundrawtransaction`:

```
fundrawtransaction "hexstring" ( options iswitness )                                        
                                              
If the transaction has no inputs, they will be automatically selected to meet its out value. 
It will add at most one change output to the outputs.
No existing outputs will be modified unless "subtractFeeFromOutputs" is specified.
Note that inputs which were signed may need to be resigned after completion since in/outputs have been added.
The inputs added will not be signed, use signrawtransactionwithkey
or signrawtransactionwithwallet for that.

...

Result:
{                     (json object)
  "hex" : "hex",      (string) The resulting raw transaction (hex-encoded string)
  "fee" : n,          (numeric) Fee in BTC the resulting transaction pays
  "changepos" : n     (numeric) The position of the added change output, or -1
}
```

In other words, w e make a basic `MsgTx` with one output, no inputs, and see what the backend would pick.